wifi: Upmerge for 2.8.0 (#18066)
nrfconnect/sdk-nrf: doc/nrf/protocols/wifi/advanced_modes/wifi_advanced_security_modes.rst

.. _ug_nrf70_wifi_advanced_security_modes:

> Wi-Fi Enterprise test: X.509 Certificate header generation

Reviewer: Seems like the text above this section is removed. Is it intentional?

Author: no, it was a gap, I fixed it now and also reworked the doc location and added an intro para.

Server certificate is also provided in the same directory for testing purposes.
Any AAA server can be used for testing purposes, for example, ``FreeRADIUS`` or ``hostapd``.

.. note::

    The certificates are for testing purposes only and should not be used in production.
    They are generated using `FreeRADIUS raddb <https://github.com/FreeRADIUS/freeradius-server/tree/master/raddb/certs>`_ scripts.

Wi-Fi PSA support
*****************

The nRF70 Series device Wi-Fi solution supports `Platform Security Architecture (PSA)`_ (PSA) APIs for cryptographic operations.

The nRF70 Series device Wi-Fi solution currently supports only WPA2-personal security profile in PSA mode.
WPA3-personal and Enterprise security profiles will be supported in future releases using PSA APIs.

Enabling Wi-Fi PSA support
==========================

To enable the Wi-Fi PSA support in your applications, you must enable the :kconfig:option:`CONFIG_WIFI_NM_WPA_SUPPLICANT_CRYPTO_ALT_NCS_PSA` Kconfig option in your application.

.. code-block:: console

    $ west build -p -b <board> samples/wifi/shell -- -DCONFIG_WIFI_NM_WPA_SUPPLICANT_CRYPTO_ALT_NCS_PSA=y

> Wi-Fi connection process is similar to the non-PSA mode, the only difference is that the cryptographic operations are performed using PSA APIs.

Reviewer: ```suggestion
The Wi-Fi connection process is similar to the non-PSA mode, however, the cryptographic operations are performed using PSA APIs.
```
